onnxruntime/core/optimizer/conv_activation_fusion.cc

```diff
@@ -56,26 +58,24 @@ bool ConvFusionDataTypeCheck(const Node& conv_node) {
     }
   }
   if (node_ep == kCpuExecutionProvider) {
-#ifdef MLAS_F16VEC_INTRINSICS_SUPPORTED
     if (!HasElementDataType(*conv_node.InputDefs()[0], ONNX_NAMESPACE::TensorProto_DataType_FLOAT) &&
-        !HasElementDataType(*conv_node.InputDefs()[0], ONNX_NAMESPACE::TensorProto_DataType_FLOAT16)) {
+        !HasElementDataType(*conv_node.InputDefs()[0], ONNX_NAMESPACE::TensorProto_DataType_FLOAT16) &&
+        support_fp16) {
       return false;
     }
-#else
-    if (!HasElementDataType(*conv_node.InputDefs()[0], ONNX_NAMESPACE::TensorProto_DataType_FLOAT)) {
-      return false;
-    }
-#endif  // MLAS_F16VEC_INTRINSICS_SUPPORTED
   }
 
   return true;
 }
 
 class ConvActivationSelector : public NodeSelector {
+ private:
+  bool support_fp16_{false};
+
  public:
   ConvActivationSelector() = default;
-
-  std::optional<NodesToOptimizeIndices> Select(const GraphViewer& graph_viewer, const Node& node) const override {
+  explicit ConvActivationSelector(bool support_fp16) : support_fp16_(support_fp16) {}
+  [[nodiscard]] std::optional<NodesToOptimizeIndices> Select(const GraphViewer& graph_viewer, const Node& node) const override {
```

Comment:
why make this one `[[nodiscard]]`? would think it makes more sense for the base class' member function to be updated. I don't think this one gets directly called.